extractor/oidc.go

```diff
@@ -59,13 +61,13 @@ func Logger(l *zap.Logger) Option {
 }
 
 // NewOIDC creates a new OIDC extractor.
-func NewOIDC(v *oidc.IDTokenVerifier, oo ...Option) (OIDC, error) {
+func NewOIDC(v *oidc.IDTokenVerifier, emailDomain string, oo ...Option) (OIDC, error) {
```

Comment:
I've tried to stick to a pattern in which only mandatory arguments (for example the `*oidc.IDTokenVerifier`) are 'first class' arguments to `NewOIDC`, with optional arguments implemented as `Option`s. Would you be willing to take a shot at copying the `Option` pattern for the email domain option?

Reply:
Yeah, that's the pattern I was suggesting. In this specific case you've already added `emailDomain` so the idea is to add an 'option' function that takes an `oidcExtractor` as an argument and sets `emailDomain`. There's a good article about the pattern [here](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis).

Something like:
```go
func EmailDomain(domain *string) Option {
	return func(o *oidcExtractor) error {
		o.emailDomain = domain
		return nil
	}
}
```

That way the following call will result in the default (i.e. an extractor `emailDomain` set to the empty string):
```go
e, err := extractor.NewOIDC(provider.Verifier(&oidc.Config{ClientID: *clientID}))
```

While this call will result in a custom email domain:
```go
e, err := extractor.NewOIDC(provider.Verifier(&oidc.Config{ClientID: *clientID}), extractor.EmailDomain("example.org"))
```